core/pkg/artifacts/saver.go

```diff
@@ -99,30 +100,65 @@ func (as *ArtifactSaver) createArtifact() (
 		runId = &as.Artifact.RunId
 	}
 
-	response, err := gql.CreateArtifact(
-		as.Ctx,
-		as.GraphqlClient,
-		as.Artifact.Entity,
-		as.Artifact.Project,
-		as.Artifact.Type,
-		as.Artifact.Name,
-		runId,
-		as.Artifact.Digest,
-		utils.NilIfZero(as.Artifact.Description),
-		aliases,
-		utils.NilIfZero(as.Artifact.Metadata),
-		utils.NilIfZero(as.Artifact.TtlDurationSeconds),
-		utils.NilIfZero(as.HistoryStep),
-		utils.NilIfZero(as.Artifact.DistributedId),
-		as.Artifact.ClientId,
-		as.Artifact.SequenceClientId,
-	)
+	// Check which fields are actually supported on the input
+	inputFieldNames, err := getInputFields(as.Ctx, as.GraphqlClient, "CreateArtifactInput")
 	if err != nil {
-		return gql.CreateArtifactCreateArtifactCreateArtifactPayloadArtifact{}, err
+		return gql.CreatedArtifactArtifact{}, err
+	}
+
+	// Note: if tags are empty, `omitempty` ensures they're nulled out
+	// (effectively omitted) in the prepare GraphQL request
+	var tags []gql.TagInput
+	if slices.Contains(inputFieldNames, "tags") {
+		for _, tag := range as.Artifact.Tags {
+			tags = append(tags, gql.TagInput{TagName: tag})
+		}
+	}
+
+	input := gql.CreateArtifactInput{
+		EntityName:                as.Artifact.Entity,
+		ProjectName:               as.Artifact.Project,
+		ArtifactTypeName:          as.Artifact.Type,
+		ArtifactCollectionName:    as.Artifact.Name,
+		RunName:                   runId,
+		Digest:                    as.Artifact.Digest,
+		DigestAlgorithm:           gql.ArtifactDigestAlgorithmManifestMd5,
+		Description:               utils.NilIfZero(as.Artifact.Description),
+		Aliases:                   aliases,
+		Tags:                      tags,
+		Metadata:                  utils.NilIfZero(as.Artifact.Metadata),
+		TtlDurationSeconds:        utils.NilIfZero(as.Artifact.TtlDurationSeconds),
+		HistoryStep:               utils.NilIfZero(as.HistoryStep),
+		EnableDigestDeduplication: true,
+		DistributedID:             utils.NilIfZero(as.Artifact.DistributedId),
+		ClientID:                  as.Artifact.ClientId,
+		SequenceClientID:          as.Artifact.SequenceClientId,
+	}
+
+	response, err := gql.CreateArtifact(as.Ctx, as.GraphqlClient, input)
+	if err != nil {
+		return gql.CreatedArtifactArtifact{}, err
 	}
 	return response.GetCreateArtifact().GetArtifact(), nil
 }
 
+func getInputFields(ctx context.Context, client graphql.Client, typeName string) ([]string, error) {
```

Comment:
This feels like it should go in another package (as a general helper func) but not sure where -- if agreed, please let me know if you have preferences for which module